src/Raven.Server/Config/Categories/IndexingConfiguration.cs

```diff
@@ -143,6 +143,13 @@ public virtual bool RunInMemory
         [ConfigurationEntry("Indexing.Analyzers.NGram.MaxGram", ConfigurationEntryScope.ServerWideOrPerDatabase)]
         public int MaxGram { get; set; }
 
+        [Description("Managed allocations limit in an indexing batch after which the batch will complete and an index will continue by starting a new one")]
+        [DefaultValue(2048)]
```

Comment:
Can this number be calculated on startup based on the total memory and the number of cores?

Reply:
The problem is "How many objects do you have", we might have enough memory to keep 10M objects but this will cause long GC pauses, which we want to avoid.
2GB is some sort of indication that this batch has generated too many objects and should be closed.

Reply:
we can adjust this number maybe by the CPU clock speed, but I'm sure we don't want to go there :-)

Reply:
We don't want to go there, yes.
The issue is that we want to limit the amount of _garbage_ we have, not to limit memory usage.
We care about GC latency more than anything else.